lib/crypt.cpp

```diff
@@ -6,6 +6,9 @@
  * Truecrypt License.
  */
 
+#ifndef _KERNEL_MODE
```

Comment:
_KERNEL_MODE should be defined in the Makefile
DEFINES = _KERNEL_MODE=1

Reply:
Maybe also add _KERNEL 
DEFINES = _KERNEL_MODE=1 _KERNEL

Reply:
Actually this should be set up by the makefile-engine already: https://cgit.haiku-os.org/haiku/tree/data/develop/makefile-engine#n43

Reply:
The problem here is that this is a static library, not a driver. It's currently actually used in the UI app, as well as the kernel driver.